mlrun/runtimes/function.py

```diff
@@ -483,6 +485,52 @@ def _wait_for_function_deployment(self, db, verbose=False):
             logger.error("Nuclio function failed to deploy", function_state=state)
             raise RunError(f"function {self.metadata.name} deployment failed")
 
+    def with_node_selection(
+        self,
+        node_name: typing.Optional[str] = None,
+        node_selector: typing.Optional[typing.Dict[str, str]] = None,
+        affinity: typing.Optional[client.V1Affinity] = None,
+    ):
+        if not mlrun.mlconf.nuclio_version:
+            raise mlrun.errors.MLRunMissingDependencyError(
+                "Missing nuclio version for node selection "
+                "compatibility verification"
+            )
+        try:
+            # TODO: handle labels like stable, unstable, latest
+            parsed_nuclio_version = semver.VersionInfo.parse(
+                mlrun.mlconf.nuclio_version
+            )
+        except ValueError as exc:
+            logger.error(
+                "Unable to parse nuclio version",
+                nuclio_version=mlrun.mlconf.nuclio_version,
+            )
+            raise exc
+
+        message = (
```

Comment:
I suggest to add a `py` file with all those version to feature relationships.
then, each function can be decorated with the feature / or supported versions

e.g.:
```
@nuclio.min_levels("1.5.20", "1.6.10")
def x():
  return "I got here after I ensured the runnning nuclio backend is either `1.5.20` or `1.6.10`
```

Reply:
I'm not a fan of function decorators, but after discussing this offline, I'm willing to give it a try
Please do put the whole version compatibility check in an helper method and only call it from the decorator (so that we can use also without the decorator)
Let's move the whole logic (decorator and helper) to be under `mlrun/runtimes/nuclio.py`

Reply:
After you move the version compatibility to another method, please add tests to it